libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -853,17 +853,59 @@ - (NSString *)scopeQueryParamString {
  - communityId
  - communityUrl
  */
-
 - (void) updateCredentials:(NSDictionary *) params {
+    NSMutableDictionary *changeSet = [NSMutableDictionary new];
+    if (params[kSFOAuthAccessToken]) {
+        if(![params[kSFOAuthAccessToken] isEqualToString:self.credentials.accessToken])
```

Comment:
Inconsistent / risky mixture of conditionals without parenthesis, combined with inline conditions...